### ISSUE TITLE
Script windows disappear a second after opening

### ISSUE THREAD

**Issue by AuthFailed** [Lailloken/Lailloken-UI#41]
Hi, I'm playing Path of Exile in Geforce Now. Not using any color filters. The first time I ran it and scanned my inventory, it worked. On subsequent launches, it just sounded like a vibration, which apparently means successful launch of the program, script windows would appear for a couple of seconds and disappear. I could not get them back. Completely reinstalling the program doesn't help.

**Comment by Lailloken** [comment]
> Completely reinstalling the program doesn't help.

Deleting the ini\config.ini file resets your settings and should start the first-time-setup. It seems to me the script can't recognize if the archnemesis inventory is open, meaning the colors of the pixels have changed slightly. Going through the first-time-setup will recalibrate the pixel check.

**Comment by AuthFailed** [comment]
Deleted ini file and first run:
![GeForceNOW_z71rJUIMwx](https://user-images.githubusercontent.com/21986465/156916797-32033476-0423-40a0-a8f8-ec5ce628e35d.jpg)
Script recognize archnemesis's inventory and shows up:
![GeForceNOW_YthqgRchmY](https://user-images.githubusercontent.com/21986465/156916815-9dc33d14-6c48-4b53-ac99-8589c0d8e4d1.jpg)
After closing and opening inventory again script doesn't show up
![GeForceNOW_7b8wo3zZcb](https://user-images.githubusercontent.com/21986465/156916821-95a34d6d-499f-4a4f-ad71-c7d714b0abb2.jpg)

**Comment by AuthFailed** [comment]
If I restart the script without deleting the configuration, there will be a vibration sound, a few blinks at the bottom left "ready" and nothing else appears.

**Comment by Lailloken** [comment]
Alright, this seems to be because of GeForce Now. The image quality must be fluctuating too much for the script to reliably check pixel colors. I have put your screenshots into a comparison tool, and the differences show in pink.
![textcompare 10_50_34 AM](https://user-images.githubusercontent.com/61888437/156918133-9f4d22d2-cef4-4d77-8cde-fd2afb8de8f9.png)
Notice how only very little of the in-game interface matches in both screenshots.

As a comparison, two of my screenshots:
![textcompare 10_52_04 AM](https://user-images.githubusercontent.com/61888437/156918158-e7f4bac2-f767-4c2b-a5c8-5f4c0baae6d1.png)

The inventory shows no difference, and for the rest I deliberately moved my character a bit to highlight it more. The squares are the animated buttons on the UI, so they are different.

Has anything changed with GeForce Now? And how long were you able to use the script before?

**Comment by AuthFailed** [comment]
First time using the script, never used it before. It's very strange that on the first run it detects every icon perfectly, but after reopening the inventory it doesn't see it.

**Comment by Lailloken** [comment]
I have never used GeForce Now, are there options for filters? Maybe it's using adaptive sharpening or contrast filters that vary depending on what's on the screen apart from UI. So moving the character (and therefore changing the camera view) might inherently change the whole image. Is there a way to force bitrate or quality?

**Comment by AuthFailed** [comment]
Да, здесь есть фильтры, но я не использую их. Битрейт и качество выставлены на самые максимальные.

**Comment by AuthFailed** [comment]
Try put this screenshot in comparison tool, please. I think it's better quality then before
https://flic.kr/p/2n79BKZ

**Comment by Lailloken** [comment]
Yeah, it's even more pink now. Not even the hp and mana orbs match now. Unfortunately, there were only two people who used GeForce Now when the script left beta, and none before that. One gave me the info I needed to implement basic support, but after that I never heard from them again, so I assumed it was just working fine.
The problem is that I could implement hotkeys to toggle the overlay, but then I don't know how well inventory recognition would work with this image variance.

**Comment by Lailloken** [comment]
If you want to troubleshoot more from your end, here's the tool [textcompare.org](https://www.textcompare.org/image/). Make sure to click "ignore nothing" in the settings.

**Comment by AuthFailed** [comment]
It's very strange that it sees everything on the first run, and then it stops. As far as I understand, it even identifies items in the inventory correctly.

**Comment by AuthFailed** [comment]
> If you want to troubleshoot more from your end, here's the tool [textcompare.org](https://www.textcompare.org/image/). Make sure to click "ignore nothing" in the settings.

How do I use it properly?

**Comment by Lailloken** [comment]
Alternatively, you can edit the allowed variance of the pixel check in the script by editing "PixelSearch, OutputVarX, OutputVarY, %x%1_x, %x%1_y, %x%1_x, %x%1_y, %x%1_color, 0, Fast RGB" in lines 1734 and 1736. It's set to 0 by default for an exact match, 255 is max.
Increase the value, restart the script, and test. Repeat until it works. The problem is the overlay might pop up randomly if similar colors are found on these pixels.

**Comment by Lailloken** [comment]
> How do I use it properly?

One screenshot goes into the first box, another one in the second. The comparison image will show up underneath and settings will show up.

**Comment by AuthFailed** [comment]
> > How do I use it properly?
> 
> One screenshot goes into the first box, another one in the second. The comparison image will show up underneath and settings will show up.

What I meant was that I didn't understand which screenshots I should compare:)

**Comment by Lailloken** [comment]
I would suggest tweaking GeForce Now settings to the point that two different screenshots with the archnemesis inventory open show as few differences as possible in the comparison image. Make sure to move the character around between the screenshots to confirm if that's where the quality variance comes from.

**Comment by AuthFailed** [comment]
I set the values on lines 1734 and 1736 to 15, and now the script detects the inventory area almost perfectly and does not trigger the other. Thank you so much for the great program and support!

**Comment by Lailloken** [comment]
Glad it works. I'll probably implement an override value for the config.ini for the next release, so that this modification carries over whenever the user updates the script and doesn't have to be done manually every time. Please report back if you find any inconsistent behavior while using it more.
